src/index.ts

```diff
@@ -241,356 +241,363 @@ export type DragOptions = {
-      // Apply user-select: none on body to prevent misbehavior
-      bodyOriginalUserSelectVal = document.body.style.userSelect;
-      document.body.style.userSelect = 'none';
-    }
+		// Dispatch custom event
+		fireSvelteDragStartEvent(node);
 
-    // Dispatch custom event
-    fireSvelteDragStartEvent(node);
+		const { clientX, clientY } = isTouchEvent(e) ? e.touches[0] : e;
 
-    const { clientX, clientY } = isTouchEvent(e) ? e.touches[0] : e;
+		if (canMoveInX) initialX = clientX - xOffset;
+		if (canMoveInY) initialY = clientY - yOffset;
 
-    if (canMoveInX) initialX = clientX - xOffset;
-    if (canMoveInY) initialY = clientY - yOffset;
+		// Only the bounds uses these properties at the moment,
+		// may open up in the future if others need it
+		if (computedBounds) {
+			clientToNodeOffsetX = clientX - nodeRect.left;
+			clientToNodeOffsetY = clientY - nodeRect.top;
+		}
+	}
 
-    // Only the bounds uses these properties at the moment,
-    // may open up in the future if others need it
-    if (computedBounds) {
-      clientToNodeOffsetX = clientX - nodeRect.left;
-      clientToNodeOffsetY = clientY - nodeRect.top;
-    }
-  }
+	function dragEnd(e: MouseEvent | TouchEvent) {
+		if (disabled) return;
+		if (!active) return;
 
-  function dragEnd(e: MouseEvent | TouchEvent) {
-    if (disabled) return;
-    if (!active) return;
+		// Apply class defaultClassDragged
+		node.classList.remove(defaultClassDragging);
+		node.classList.add(defaultClassDragged);
 
-    // Apply class defaultClassDragged
-    node.classList.remove(defaultClassDragging);
-    node.classList.add(defaultClassDragged);
+		if (applyUserSelectHack) document.body.style.userSelect = bodyOriginalUserSelectVal;
 
-    if (applyUserSelectHack) document.body.style.userSelect = bodyOriginalUserSelectVal;
+		fireSvelteDragStopEvent(node);
 
-    fireSvelteDragStopEvent(node);
+		if (canMoveInX) initialX = translateX;
+		if (canMoveInX) initialY = translateY;
 
-    if (canMoveInX) initialX = translateX;
-    if (canMoveInX) initialY = translateY;
+		active = false;
+	}
 
-    active = false;
-  }
+	function drag(e: TouchEvent | MouseEvent) {
+		if (!active) return;
 
-  function drag(e: TouchEvent | MouseEvent) {
-    if (!active) return;
+		// Apply class defaultClassDragging
+		node.classList.add(defaultClassDragging);
 
-    // Apply class defaultClassDragging
-    node.classList.add(defaultClassDragging);
+		e.preventDefault();
 
-    e.preventDefault();
+		nodeRect = node.getBoundingClientRect();
 
-    nodeRect = node.getBoundingClientRect();
+		const { clientX, clientY } = isTouchEvent(e) ? e.touches[0] : e;
 
-    const { clientX, clientY } = isTouchEvent(e) ? e.touches[0] : e;
+		// Get final values for clamping
+		let [finalX, finalY] = [clientX, clientY];
 
-    // Get final values for clamping
-    let [finalX, finalY] = [clientX, clientY];
+		// Calculate the current scale of the node
+		let scale = node.offsetWidth / nodeRect.width;
```

Comment:
Can you explain what this line does? What's the difference between offsetWidth and nodeRect.width?

Reply:
> Help me understand this: Why scale is needed in the first place? Is there some unexpected behavior that happens without it?

Imagine that I have a canvas where I have draggable elements on top of it (let's say we have element A 100px by 100px), and I want to implement a zoom in and out function. I can add `transform: scale(2);` to zoom in 2x. But that breaks the calculation as it'll still change the translation by 100px but it actually needs to change it by 50px only as the scale is 2x. 

In the current behaviour the draggable element will either move too fast (if zoomed in) or move too slow (if zoomed out) compared to the cursor. This breaks the mouse events logic (mouseup happens outside the element) and also looks weird for the user.

`node.offsetWidth` gives you the unscaled width and the `nodeRect` gives you the transformed width. By using those I can calculate the scale to be used in calculation of translate values. When scale is added draggable element moves at the speed of the cursor when scaled, fixing the problems.

Reply:
This is perfect! Though I'll be renaming scale to `inverseScale`, as scale to me feels like "the size compared to the original size", so just this!

Thanks for the PR